advanced_alchemy/config/common.py

```diff
@@ -117,6 +117,8 @@ class GenericSQLAlchemyConfig(Generic[EngineT, SessionT, SessionMakerT]):
 
     If set, the plugin will use the provided instance rather than instantiate an engine.
     """
+    create_all: bool = False
+    """If true, all models are automatically created on engine creation."""
```

Comment:
```suggestion
    create_all: bool | MetaData = False
    """If true, or an instance of ``MetaData``, all models are automatically created on engine creation.
    
    If ``True``, it is assumed you are using one of the advanced-alchemy base models and that metadata is used.
    
    If a ``MetaData`` instance is passed, it is used to create the schema.
    """
```
WDYT about that functionality? This makes it a bit more flexible. From the PoV of the Litestar SQLAlchemy tutorial, it constructs its own `Base` so it would allow that metadata to be passed in.

Reply:
So, there is a `metadata` attribute on the `alembic` config.  By default, it uses the built in metadata base, but you can override if you'd like.  This is the metadata that I'm fetching for the creation here.

That said, I like the idea of the make it easier to pass in a custom metadata, is there a way we can consolidate between the alembic config and this one?

Reply:
I'm not overly familiar with the alembic stuff in here yet, what about something like this?

```diff
diff --git a/advanced_alchemy/config/common.py b/advanced_alchemy/config/common.py
index 2c25b3f..c09a6c8 100644
--- a/advanced_alchemy/config/common.py
+++ b/advanced_alchemy/config/common.py
@@ -84,6 +84,11 @@ class GenericSessionConfig(Generic[ConnectionT, EngineT, SessionT]):
 class GenericSQLAlchemyConfig(Generic[EngineT, SessionT, SessionMakerT]):
     """Common SQLAlchemy Configuration."""
 
+    alembic_config: GenericAlembicConfig
+    """Configuration for the SQLAlchemy Alembic migrations.
+    
+    The configuration options are documented in the Alembic documentation.
+    """
     create_engine_callable: Callable[[str], EngineT]
     """Callable that creates an :class:`AsyncEngine <sqlalchemy.ext.asyncio.AsyncEngine>` instance or instance of its
     subclass.
@@ -117,14 +122,25 @@ class GenericSQLAlchemyConfig(Generic[EngineT, SessionT, SessionMakerT]):
 
     If set, the plugin will use the provided instance rather than instantiate an engine.
     """
+    metadata: MetaData | None = None
+    """Optional metadata to use.
+    
+    If set, the plugin will use the provided instance rather than the default metadata. 
+    """
     create_all: bool = False
-    """If true, all models are automatically created on engine creation."""
+    """If true, all models are automatically created on engine creation.
+    
+    If metadata is provided, it is used to create the schema, otherwise the default metadata is used.
+    """
 
     def __post_init__(self) -> None:
         if self.connection_string is not None and self.engine_instance is not None:
             msg = "Only one of 'connection_string' or 'engine_instance' can be provided."
             raise ImproperConfigurationError(msg)
 
+        if self.metadata:
+            self.alembic_config.target_metadata = self.metadata
+
```

Reply:
This should work just fine!

Reply:
@peterschutt take a look at the changes.  I think this is what we want.